RFC Draft.md

```diff
@@ -535,6 +535,111 @@ function subscribeToCreate() {
 }
 ```
 
+## Flatten the GraphQL operations' responses
+As alluded to in the prior suggestion, we're looking to flatten the results for the GraphQL operations to make them more easily accessible instead of the current three-levels-deep nested object. Would love to get your understanding on which option you prefer.
+
+**Current Usage (v5)**
+```ts
+async function createNewTodo() {
+  const res: GraphQLResult<Todo> = await API.graphql<GraphQLQuery<Todo>>(
+    graphqlOperation(createTodo, {
+      input: { id: uuid() },
+    })
+  );
+  // todo result is nested
+  console.log(res.data.createTodo); 
+}
+
+interface GraphQLResult<T = object> {
+    data?: T; // in the above example T is Todo
+    errors?: GraphQLError[];
+    extensions?: {
+        [key: string]: any;
+    };
+}
+```
+
+### Proposed behavior for single query/mutation in the response
+**Proposed Option 1: Flatten to the lowest level (v6)**
+```ts
+async function createNewTodo() {
+  const res: Todo = await API.graphql<GraphQLQuery<Todo>>(
+    graphqlOperation(createTodo, {
+      input: { id: uuid() },
+    })
+  );
+  
+  // res flattened to todo level
+  console.log(res); 
+}
+```
+
+**Proposed Option 2: Flatten to the `data` level (v6)**
+```ts
+async function createNewTodo() {
+  const res: GraphQLData<Todo> = await API.graphql<GraphQLQuery<Todo>>(
+    graphqlOperation(createTodo, {
+      input: { id: uuid() },
+    })
+  );
+  // res in at .data level
+  console.log(res.createTodo); 
+}
+
+interface GraphQLData<T = object> {
+  [query: string]: T // in the above example T is Todo
+}
```

Comment:
Makes sense to leave this type in I think, explains what the `createTodo` actually is